### PR TITLE
ReadRows acceptance tests

### PR DESCRIPTION
**Pull request by evildour**
This revealed a few bugs in the row merging logic as well.

### CODE REVIEW

**Comment by evildour** [comment]
Oops, I forgot the last ItemGroup in the project file was auto-generated. Fixing now.

**Comment by evildour** [comment]
Not sure why a bunch of these tests are failing. Looking into it.

**Comment by evildour** [comment]
Figures it would be a last minute change which I thought was harmless, so I didn't run the tests again :/

Also, there are two tests which are legitimately failing and should have on my local machine, but weren't before and are now...no clue what's going on there.

File: apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2.Tests/ReadRowsAcceptanceTest.cs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        public async Task Test(ChunkTestCase testCase)
+        {
+            var stream = new MockReadRowsStream(
+                new ReadRowsResponse
+                {
+                    Chunks = { testCase.Chunks.Select(json => CellChunk.Parser.ParseJson(json)) }
+                });
+
+            var responses = new List<Row>();
+            try
```
Comment (chrisdunelm) [inline apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2.Tests/ReadRowsAcceptanceTest.cs]: I'd rather see this written more explicitly. I.e. `if (testCase.ExpectsError) Assert.Throws...
To my mind this makes the intent clearer, and makes use of the more suitable Assert statement.

File: apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2.Tests/ReadRowsAcceptanceTest.cs [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+                              RowKey = row.Key.ToStringUtf8(),
+                              FamilyName = family.Name,
+                              Qualifier = column.Qualifier.ToStringUtf8(),
+                              Timestamp = cell.TimestampMicros,
+                              Value = cell.Value.ToStringUtf8(),
+                              Label = cell.Labels.FirstOrDefault() ?? "",
+                              Error = false
+                          };
+            var expectedResults = testCase.GetNonExceptionResults();
+            Assert.True(expectedResults.SequenceEqual(results), testCase.Name);
```
Comment (chrisdunelm) [inline apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2.Tests/ReadRowsAcceptanceTest.cs]: Assert.Equals(...) works fine on sequences. SequenceEqual is not required.

File: apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/RowAsyncEnumerator.cs [hunk 3]
```diff
@@ -96,6 +95,14 @@ IEnumerator<Row> MergeResponseChunks(ReadRowsResponse response)
                 {
                     if (chunk.ResetRow)
                     {
+                        Assert(
```
Comment (chrisdunelm) [inline apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/RowAsyncEnumerator.cs]: I realise `Assert` is already in the code; but why aren't we using `GaxPreconditions...` instead?

File: apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2.Tests/ReadRowsAcceptanceTest.cs (hunk 1 above)
Comment (evildour) [reply apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2.Tests/ReadRowsAcceptanceTest.cs]: Done

File: apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2.Tests/ReadRowsAcceptanceTest.cs (hunk 2 above)
Comment (evildour) [reply apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2.Tests/ReadRowsAcceptanceTest.cs]: Unfortunately, `Assert.Equals` doesn't allow a failure message to be provided, so if there's a failure on the build/CI machines, there would be no way to know which case failed.

File: apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/RowAsyncEnumerator.cs (hunk 3 above)
Comment (evildour) [reply apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/RowAsyncEnumerator.cs]: We need to perform additional work on failure: reset the row merger's state.

**Pull request merged**